### DISCUSSION TITLE
How to detect client device type (computer or mobile)?

### DISCUSSION THREAD

**Discussion by happybeginning1** [zauberzeug/nicegui#3301] [Q&A]
### Question

Hi everyone,

I'm working on a project and need to find a way for my program to monitor the type of device the client is using, such as whether it's a computer or a mobile device. Does anyone have any suggestions or methods to achieve this?

Thanks in advance!

**Comment by python-and-novella** [comment]
get user-agent,and try to split it.
```python3
app.on_connect(lambda :print( [i[1] for i in ui.context.client.environ['asgi.scope']['headers'] if i[0]==b'user-agent']))
```

**Comment by rodja** [comment]
With 1.4.28 we introduced the `request` property on the `client`:

```py
app.on_connect(lambda: print(ui.context.client.request.headers['user-agent']))
```

**Comment by happybeginning1** [comment]
Thanks @python-and-fiction @rodja,
I have sovled this issue with your help.
Thank you!
